Composer/packages/extensions/intellisense/src/components/CompletionElement.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { CompletionItemKind } from 'monaco-languageclient';
+import { FontIcon } from 'office-ui-fabric-react/lib/Icon';
+import { DirectionalHint, TooltipHost } from 'office-ui-fabric-react/lib/Tooltip';
+import React, { CSSProperties } from 'react';
+import { CompletionItem, MarkupContent } from 'vscode-languageserver-types';
+
+type FuseJsMatch = { indices: number[][]; value: string; key: string };
+
+const styles: Record<string, CSSProperties> = {
+  completionElement: {
+    height: '32px',
+    cursor: 'pointer',
+    padding: '0 4px',
+    width: '100%',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  selectedElement: { backgroundColor: '#ddd' },
+  text: { fontSize: '15px' },
+  icon: { marginRight: '5px' },
+};
+
+const renderIcon = (kind: CompletionItemKind | undefined): JSX.Element => {
+  let icon: JSX.Element = <> </>;
+
+  switch (kind) {
+    case CompletionItemKind.Function:
+      icon = <FontIcon iconName="Variable" style={styles.icon} />;
+      break;
+    case CompletionItemKind.Variable:
+      icon = <FontIcon iconName="VariableGroup" style={styles.icon} />;
+      break;
+    case CompletionItemKind.Enum:
+      icon = <FontIcon iconName="BulletedList" style={styles.icon} />;
+      break;
+  }
+
+  return icon;
+};
+
+const renderLabelWithCharacterHighlights = (matches: FuseJsMatch[]): JSX.Element => {
+  const renderMatch = (match: FuseJsMatch, segmentIndex: number): JSX.Element => {
+    let firstIndex = 0;
+    const lastIndex = match.value.length;
+
+    const items = match.indices.map((m, spanIndex) => {
+      const firstSpan = <span>{match.value.slice(firstIndex, m[0])}</span>;
+      const secondSpan = <span style={{ color: 'blue' }}>{match.value.slice(m[0], m[1] + 1)}</span>;
```

Comment:
Are you certain that there is a m[0] and m[1]?

Reply:
Yes, based on the returned results from a FUSE.js search.

Reply:
It is worth a comment to say that Fuse searches always return at least 2 items then.  Otherwise, anyone reading this code fresh will have the same concern.